components/net/fetch/methods.rs

```diff
@@ -732,9 +737,29 @@ async fn scheme_fetch(
                 if let Ok(file) = File::open(file_path.clone()) {
                     if let Ok(metadata) = file.metadata() {
                         if metadata.is_dir() {
-                            return Response::network_error(NetworkError::Internal(
-                                "Opening a directory is not supported".into(),
-                            ));
+                            let url = if !url.path().ends_with('/') {
```

Comment:
Generally speaking, I think we should move all the code to `components/servo/`, and only reach it when an user manually inputs a directory(or file?) in the address bar. We don't want a script using `fetch` being able to read the user's file system, I think.

Reply:
As discussed in zulip, I now think we can keep the code in `net`, but perhaps outside of `fetch`. See the existing `FileManager`.